universe/Universe.cpp

```diff
@@ -1660,6 +1660,21 @@ void Universe::ForgetKnownObject(int empire_id, int object_id) {
     objects.Remove(object_id);
 }
 
+void Universe::CheckShipDesignVisibility(int empire_id, int object_id, Visibility vis) {
```

Comment:
It's not my understanding that a verb form function name carries a contract or even implication that there can be no constraints on the performance of the action, but I won't press that point.

A name like MakeDesignVisibileToEmpireIfShipAtLeastPartiallyVisible seems to me to put excessive implementation detail in the name.  As for separating the visibility test, that just seems like extra code duplication to me.  

I'll make another proposal: HandleShipDesignKnowledgeGrant